src/extension/sldService/src/test/java/org/geoserver/sldservice/rest/ClassifierTest.java

```diff
@@ -1987,4 +1992,134 @@ private void checkRuleLineSymbolizer(Rule rule, String color) {
         assertNotNull(symbolizer.getStroke());
         assertEquals(color, symbolizer.getStroke().getColor().toString());
     }
+
+    @Test
+    public void testOpenIntervalFirstRuleConsistency() throws Exception {
+        final String restPath =
+                RestBaseController.ROOT_PATH
+                        + "/sldservice/cite:ClassificationPoints2/"
+                        + getServiceUrl()
+                        + ".xml?"
+                        + "attribute=bar&ramp=red&method=quantile&intervals=2&open=true";
+        Document dom = getAsDOM(restPath, 200);
+        print(dom);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        print(dom, baos);
+        String resultXml = baos.toString().replace("\r", "").replace("\n", "");
+        Rule[] rules =
+                checkSLD(resultXml.replace("<Rules>", sldPrefix).replace("</Rules>", sldPostfix));
+        assertTrue(rules.length == 2);
+        assertTrue(rules[0].getFilter() instanceof PropertyIsEqualTo);
```

Comment:
I'd also check that the second rule here, and that they won't overlap?

Reply:
fixed